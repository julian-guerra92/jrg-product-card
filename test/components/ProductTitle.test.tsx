import React from 'react';
import render from 'react-test-renderer'
import { ProductTitle } from '../../src/components/ProductTitle';
import { ProductCard } from '../../src/components/ProductCard';
import { product1 } from '../data/productos';

describe('Pruebas en el ProducTitle', () => {

   test('Debe de mostrar el componenete correctamente con el tÃ­tulo personalizado', () => {
      const wraper = render.create(
         <ProductTitle title='Custom Product' />
      )
      expect(wraper.toJSON()).toMatchSnapshot();
   })

   test('Debe de mostrar el componenete con el nombre del producto', () => {
      const wraper = render.create(
         <ProductCard product={product1}>
            {
               () => (
                  <ProductTitle />
               )
            }
         </ProductCard>
      )
      expect(wraper.toJSON()).toMatchSnapshot();
   })


})
