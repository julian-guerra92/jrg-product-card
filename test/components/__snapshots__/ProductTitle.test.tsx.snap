// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el ProducTitle Debe de mostrar el componenete con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Sin imagen
  </span>
</div>
`;

exports[`Pruebas en el ProducTitle Debe de mostrar el componenete correctamente con el t√≠tulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product
</span>
`;
